Fix STM32G0B/C DMA channel count

chris-durand opened the pull request:
The device data incorrectly indicates up to 12 channels while the peripheral only supports up to 7.

![dma_g0](https://github.com/modm-io/modm-devices/assets/25187160/e7935e7b-b756-424e-9cf2-6d610161d61f)

chris-durand committed:
Update STM32G0 device files

salkinium commented on tools/generator/dfg/stm32/stm_device_tree.py:
```diff
@@ -390,9 +390,15 @@ def deduplicate_list(l):
+                    # STM32G0B/C device data is wrong, only max 7 channels are available
+                    if did.family == 'g0' and did.name in ['b0', 'b1', 'c1'] and channel > 7:
+                        continue
+
```
Do you perhaps want to patch the CubeMX database instead? There's a mechanism for it, and I'm trying to reduce the amount of magic code.

chris-durand replied:
Done. Changes are pushed, but don't appear here. github has a major outage of pull requests and actions right now.

chris-durand replied:
Github is working again. The pipeline has passed.

chris-durand committed:
[dfg] Fix STM32G0B/C max DMA channel count in device data

The device data indiciates 12 channels for instance 1 but only 7
are actually supported.

salkinium reviewed (approved):
Thanks!
